### DISCUSSION TITLE
Additional inference

### DISCUSSION THREAD

**Discussion by gschuager** [nrwl/nx#18907] [Feature Requests]
I'm looking into how to infer other project properties besides targets, mainly project name and project type (app vs lib).

Is this supported in any way? If not, do you think this is something worth to add as a new feature?

Basically, I'm trying to avoid putting as much content as possible in my project.json and instead to derive it using conventions in a workspace plugin, but I'm still forced to explicitly put both the project name and project type in that file.

**Comment by AgentEnder** [comment]
Yeah, this is supported in the plugin API v2. Not in the `projectFilePatterns` + `registerProjectTargets` api (v1) though. 

Docs are incoming: https://github.com/nrwl/nx/pull/18890

**Reply by gschuager** [reply]
Nice, thanks, looking forward to this.
